packages/css/src/components/file-input/file-input.scss

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+@import "../../common/text-rendering";
+
+@mixin reset-button {
+  border: 0;
+}
+
+.ams-file-input {
+  align-items: center;
+  color: var(--ams-file-input-color);
+  display: flex;
+  font-family: var(--ams-file-input-font-family);
+  font-size: var(--ams-file-input-font-size);
+  font-weight: var(--ams-file-input-font-weight);
+  justify-content: flex-start;
+  line-height: var(--ams-file-input-line-height);
+  outline-offset: var(--ams-file-input-outline-offset);
+  touch-action: manipulation;
+  width: 100%;
+
+  @include text-rendering;
+}
+
+.ams-file-input:disabled {
+  color: var(--ams-file-input-disabled-color);
+  cursor: not-allowed;
+}
+
+.ams-file-input::file-selector-button {
+  background-color: var(--ams-file-input-file-selector-button-background-color);
+  box-shadow: var(--ams-file-input-file-selector-button-box-shadow);
+  color: var(--ams-file-input-file-selector-button-color);
+  cursor: var(--ams-file-input-file-selector-button-cursor);
+  display: inline-block;
+  font-family: inherit;
+  font-size: inherit;
+  font-weight: inherit;
+  /* stylelint-disable-next-line property-disallowed-list */
+  margin-right: 2ex; // Add space between the file selector button and the file name
+  padding-block: var(--ams-file-input-file-selector-button-padding-block);
+  padding-inline: var(--ams-file-input-file-selector-button-padding-inline);
+
+  @include reset-button;
+}
+
+.ams-file-input:disabled::file-selector-button {
+  background-color: var(--ams-file-input-file-selector-button-disabled-background-color);
+  box-shadow: var(--ams-file-input-file-selector-button-disabled-box-shadow);
+  color: var(--ams-file-input-disabled-color);
+  cursor: var(--ams-file-input-file-selector-button-disabled-cursor);
```

Comment:
This is unnecessary, the disabled cursor is already defined on the input.

Reply:
`::file-selector-button` is kind of an oddball, it does not inherit rules from the parent.

Reply:
Ha, something weird happens here. You forgot to define `--ams-file-input-file-selector-button-disabled-cursor`, but somehow having an undefined var makes `file-selector-button` inherit this value from the input? Weird. 

Anywhoo, you're right, you have to specifically set the cursor. Can you define the var as well?

Reply:
Var is defined